### ISSUE TITLE
Different escape strategies and italic interpretation (on different operation systems and AdaptiveCard elements)

### ISSUE THREAD

**Issue by sueess** [MicrosoftDocs/msteams-docs#5793]
Hi there

I know the title is hard to understand, but it exactly shows the mess :). Let's start with the problems:

1. In the documentation it says to use _italic_ to display text italic. On some devices/OS this *italic* also works. On some others it does not. This is very problematic, since we can not distinguish between OS when we send a message. **Solution: Make the interpretation everywhere the same.**
2. Since the * is still working on some devices, we are escaping it like this \*. Like this we get also different representations. Android an iOS show the backslash, OSX and Windows not. **Solution: Make the interpretation everywhere the same.**
3. The Markdown is not working consistently in all AdaptiveCard selectors. E.g. in Input.ChoiceSet the Markdown is not accepted. Why not just making things consistent in all elements?  **Solution: Make the interpretation everywhere the same.**

In short: Please ensure that Markdown and Escaping works on all devices, AdaptiveCard elements and OS the same way

![macOS](https://user-images.githubusercontent.com/6587992/165261100-11cea909-5416-4f7f-9ed2-a85a3748224e.png)
![android](https://user-images.githubusercontent.com/6587992/165261108-232a1805-72a0-4b0a-919f-1b2f8c51760e.jpg)
![iOS-mobile](https://user-images.githubusercontent.com/6587992/165261112-66808886-6b9b-4ada-b783-759d7de90253.png)
![Windows](https://user-images.githubusercontent.com/6587992/165261462-64704ce2-577b-4ce1-a4f0-4c6437cd75a7.png)
.

**Comment by sayali-MSFT** [comment]
@sueess -We are looking into this I will get back to you soon.

**Comment by sayali-MSFT** [comment]
@sueess -We are tried to repro the scenario with IOS and android device but markdown is working fine in both.
Could you please confirm with which device getting issue?
IOS Device -
![MicrosoftTeams-image (27)](https://user-images.githubusercontent.com/93527375/165517845-055c9b09-52f8-4b93-9e34-bc6af8497ecd.png)
Android Device -
![MicrosoftTeams-image (29)](https://user-images.githubusercontent.com/93527375/165518558-84b3e090-4770-4419-a042-55b660708ed1.png)
Windows:-
![image](https://user-images.githubusercontent.com/93527375/165520171-7278495d-ed32-47d8-b9e5-a615dc0a9396.png)

**Comment by sueess** [comment]
Thanks for looking into that problem. We tested again and here is the most actual documentation. I hope it is understandable.

**Adaptive card using * markdown for italic**

- Facts
    - Same behaviour on mobile and windows/osx
    - Different behaviour in TextBlock and input.ChoiceSet
- Suggestion
    - Same behaviour in all elements 

```
{
  type: 'AdaptiveCard',
  body: [
    {
      type: 'TextBlock',
      text: 'Text with stars to make it italic: *text*',
      wrap: true
    },
    {
      type: 'TextBlock',
      text: 'This is a *. Sometimes we like to use a * in the text, or in the middle of a wor*d.',
      wrap: true
    },
    {
      type: 'Input.ChoiceSet',
      id: 'multipleCardSelection',
      style: 'expanded',
      choices: [
                { 
                    title: 'Text with stars to make it italic: *text*', 
                    value: 1
                },
            {
                title: 'This is a *. Sometimes we like to use a * in the text, or in the middle of a wor*d.',
                value: 2
            }
      ],
      isMultiSelect: true,
      wrap: true
    }
  ],
  actions: [],
  '$schema': 'http://adaptivecards.io/schemas/adaptive-card.json',
  version: '1.4'
}
```

![mobile](https://user-images.githubusercontent.com/6587992/165726153-e6c3d567-d7d4-48b4-b4c0-b55095d07791.png)
![windows](https://user-images.githubusercontent.com/6587992/165726183-3f411e52-09b9-4be5-8e44-289f830625db.png)


**Adaptive card using * in text**

- Facts
    - Since * is interpretated as markdown (see example above) we need to escape the * if we wanna use it in text
    - Different behaviour in TextBlock and input.ChoiceSet
    - Different behaviour on Mobile and Windows/OSX
- Suggestion
    - Same behaviour in all elements
    - Same behaviour on all OS

```
{
  type: 'AdaptiveCard',
  body: [
    {
      type: 'TextBlock',
      text: 'Text with stars to make it italic: \\*text\\*',
      wrap: true
    },
    {
      type: 'TextBlock',
      text: 'This is a \\*. Sometimes we like to use a \\* in the text, or in the middle of a wor\\*d.',
      wrap: true
    },
    {
      type: 'Input.ChoiceSet',
      id: 'multipleCardSelection',
      style: 'expanded',
      choices: [
            { 
                    title: 'Text with stars to make it italic: \\*text*', 
                    value: 1
                },
            {
                title: 'This is a \\*. Sometimes we like to use a \\* in the text, or in the middle of a wor\\*d.',
                value: 2
            }
      ],
      isMultiSelect: true,
      wrap: true
    }
  ],
  actions: [],
  '$schema': 'http://adaptivecards.io/schemas/adaptive-card.json',
  version: '1.4'
}
```

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/6587992/165727053-5addff3b-5bd8-4eea-9941-d8b192709b54.png)
![MicrosoftTeams-image](https://user-images.githubusercontent.com/6587992/165727058-21aadfda-f208-47a5-83e4-475952169e44.png)

**Comment by sayali-MSFT** [comment]
@sueess -Currently markdown feature supports TextBlock. [TextBlock](https://adaptivecards.io/explorer/TextBlock.html) offers advanced features for formatting and localizing the text.
https://docs.microsoft.com/en-us/adaptive-cards/authoring-cards/text-features#markdown-example

![markdown](https://user-images.githubusercontent.com/93527375/165733960-053c2dc5-6869-4694-99fa-3e863ac21878.png)

**Comment by sayali-MSFT** [comment]
@sueess -ok Got it. We are able to repro that issue , We have raised bug for the same and concerned team is looking into it. We will inform you once we get any update.

**Comment by sueess** [comment]
Nice! Thank you!

**Comment by sueess** [comment]
Is there any update?

**Comment by sayali-MSFT** [comment]
Engineering team working on this issue. Currently we don’t have ETA or Update on this thread. 
We will update this thread once the bug is fixed.

**Comment by sayali-MSFT** [comment]
Engineering team working on it will take time to rolled out publicly.
We will update this thread once publicly available.

**Comment by MajaEffenberg** [comment]
Is there any update for this bug?
We still have this problem, that Markdown and Escaping is displayed in different way on different devices.

The same Adaptive Card displayed in Microsoft Teams Windows App and Android App:
![image-20221102-075606](https://user-images.githubusercontent.com/89073193/199796745-17f8ed10-6c86-4457-b604-5031a8aa2add.png)
![Screenshot_20221102_085330 (1)](https://user-images.githubusercontent.com/89073193/199796736-1df679f3-607b-417a-936a-bb00bfa0906d.jpg)

**Comment by sayali-MSFT** [comment]
@MajaEffenberg -Issue is solved on latest canary build. We will update you once it is publicly released.

**Comment by sayali-MSFT** [comment]
Sorry for delay in response!!
We are actively checking the status of bug with engineering team and let you know the updates, if we have any.

**Comment by botch8** [comment]
Is there an update on this? We are hitting this as well.

**Comment by sayali-MSFT** [comment]
@MajaEffenberg @botch8 -Hope you are doing well!!
Just wanted to inform you that engineering team is working on this issue, and we are tracking it closely.
Once we get any update on it, we will inform you accordingly.

**Comment by sschoeb** [comment]
@sayali-MSFT The engineering team is now working for 3 months on this. Do you have any additional information on this?

**Comment by Prasad-MSFT** [comment]
@sueess, we got below update from engineering team:

1. [By spec](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-format?tabs=adaptive-md%2cconnector-html#format-cards-with-markdown), Adaptive Card only supports Markdown for TextBlock and Fact / FactSet, while the other fields should remain pure text. So, for MultiChoice there's no need to escape markdown but just keep text straightforwardly.
![image](https://github.com/user-attachments/assets/df3b70c2-d8fb-4db3-9290-3aa7bc1d5afd)
**(Same behaviour in Desktop/Web and Mobile clients)**



2. For escaping in Markdown in TextBlock:
![image](https://github.com/user-attachments/assets/e0d0d7e6-4e99-4c87-bd2d-8eb1297cc34d)
**(Same behaviour in Desktop/Web and Mobile clients)**

**Comment by sschoeb** [comment]
@Prasad-MSFT it took you 10 months to have an answer and you expect us to verify your answer within 6 days before the ticket is closed?

I cannot yet tell you if your response solves our problem, but I'll test it as fast as possible.
